Library/Homebrew/formula_creator.rb

```diff
@@ -56,7 +49,9 @@ def head?
       @head || args.HEAD?
     end
 
-    def generate!
+    def write_formula!
+      raise "name should not be empty" if @name.to_s == ''
+      path = @tap.new_formula_path(@name)
```

Comment:
```suggestion
      raise ArgumentError, "name is blank!" if @name.blank?
      raise ArgumentError, "tap is blank!" if @tap.blank?
      
      path = @tap.new_formula_path(@name)
```

Reply:
The @tap can not be empty. It is checked here.

https://github.com/Homebrew/brew/blob/1b8207bf47730b6fd328ac54d59e7baa635e0b62/Library/Homebrew/dev-cmd/create.rb#L153

Also there is `undefined method `blank?' for nil:NilClass`

https://docs.ruby-lang.org/en/master/NilClass.html

Reply:
@abitrolly That relies on the caller not being changed in future. ActiveSupport adds `blank?` to `NilClass`.

Gently, things are going to be a lot quicker if you stop arguing with me here, thanks.